File: Client-Side/src/app/ComputerComponent.ts

export class ComputerComponent{
    category:string;
    name:string;
    brand:string;
    productLine:string;
    numOfCores:string;
    processorClockSpeed:string;
    graphicClockSpeed:string;
    size:string;
    dimension:string;
    intrfce:string;
    colour:string;
    resolution:string;
    price:number;
    quantity:number;
    uuid:string;
    numInBasket:number;
    constructor(
        category:string,
        name:string,
        brand:string,
        productLine:string,
        numOfCores:string,
        processorClockSpeed:string,
        graphicClockSpeed:string,
        size:string,
        dimension:string,
        intrfce:string,
        colour:string,
        resolution:string,
        price:number,
        quantity:number,
        uuid:string
        ){
this.category = category;
this.name = name;
this.brand = brand;
this.productLine = productLine;
this.numOfCores = numOfCores;
this.processorClockSpeed = processorClockSpeed;
this.graphicClockSpeed = graphicClockSpeed;
this.size = size;
this.dimension = dimension;
this.intrfce = intrfce;
this.colour=colour;
this.resolution = resolution;
this.price = price;
this.quantity=quantity;
this.uuid = uuid;
this.numInBasket=0;
    }

    public getUuid():string{
        return this.uuid;
    }
    public getNumInBasket():number{
        return this.numInBasket;
    }
}

File: Server-Side/src/main/java/com/server/side/components/ComputerComponent.java
package com.server.side.components;

import java.util.Objects;
import java.util.UUID;

import javax.persistence.*;


@Entity
@Table
public class ComputerComponent {

    private static final String NOT_APPLICABLE = "N/A";

    @Id
    @Column(name = "uuid", nullable = false)
    private UUID uuid;
    @Column(name = "CATEGORY", nullable = false)
    private String category;
    @Column(name = "NAME", nullable = false)
    private String name;
    @Column(name = "BRAND", nullable = false)
    private String brand;
    @Column(name = "PRICE", nullable = false)
    private int price;
    @Column(name = "QUANTITY", nullable = false)
    private int quantity;

    public ComputerComponent() {

    }

    public ComputerComponent(Builder builder) {
        this.uuid = builder.uuid;
        this.category = builder.category;
        this.name = builder.name;
        this.brand = builder.brand;
        this.price = builder.price;
        this.quantity = builder.quantity;
    }

    public UUID getUuid() {
        return uuid;
    }

    public String getCategory() {
        return category;
    }

    public String getName() {
        return name;
    }

    public String getBrand() {
        return brand;
    }

    public int getPrice() {
        return price;
    }

    public int getQuantity() {
        return quantity;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof ComputerComponent)) {
            return false;
        }
        ComputerComponent that = (ComputerComponent) o;
        return (price == that.price) && (quantity == that.quantity) && Objects.equals(uuid, that.uuid) && Objects.equals(category, that.category) && Objects.equals(name, that.name) && Objects.equals(brand, that.brand);
    }

    public void setQuantity(int quantity) {
        this.quantity = quantity;
    }

    public static class Builder {

        protected UUID uuid;
        protected String category;
        protected String name;
        protected String brand;
        protected int price;
        protected int quantity;

        public Builder() {
        }
    }
}
